text/0031-transactions-with-contingent-inputs.md

```diff
@@ -0,0 +1,191 @@
+
+# Motivation
+[motivation]: #motivation
+
+Contingent inputs are interesting because they allow users to construct "transaction fragments" which can then be sent to other users safely.
+For example, if I want to buy a token for 5 MOB, I could create an output which gives me that token, and sign a contingent input which spends 5 of my MOB
+in exchange for someone producing that output. I can then bundle this up and send it to a counter party -- if they agree to trade at my price, they can build
+a transaction using my transaction fragment and submit it.
+
+There is no way that they can spend my 5 MOB without producing the desired output for me, so my funds
+are safe, and moreover, never leave my custody during this process until the moment the trade is final.
+It costs me nothing to produce a transaction fragment and does not require an on-chain event -- there is only one actual on chain transaction
+to settle the swap, which happens with the speed and privacy of MobileCoin.
+
+The broader significance is the following:
+* If MCIP #25 happens, then MobileCoin will become a multi-asset ecosystem. However, there will not be simple ways for the multiple tokens
+  to interact on chain at that point.
+* In cryptocurrency ecosystems, it is important to provide users with convenient ways to swap tokens of various types.
+
+In the Ethereum ecosystem, the dominant way of doing this is using tools like Automated Market Makers (AMM's). But, this requires smart contract
+functionality, which is very complex. It is also quite difficult to create AMM's with significant privacy properties.
+
+Many research papers have been written on cross-chain atomic swaps, particularly, describing protocols using Bitcoin's scripting language
+to permit swapping bitcoin in a "trustless" way for other assets, using hash time locked contracts. These protocols are often quite complex,
+requiring each party to "commit" their funds first, then trigger the swap, and have contingencies for each party to recover their committed funds
+if the counter party renegs. These protocols may require several rounds of interaction, and require several on-chain transactions, incurring fees and
+increasing the total delay for the swap to be finished.
+
+This proposal is considerably simpler -- the rounds of interaction are a minimum, requiring only one side to send a transaction fragment to another.
+No one is required to "escrow" their funds into a smart contract or temporary address. A party who desires to swap can broadcast their transaction fragment
+to many potential counterparties. Because the Ring MLSAG contains the key image of the spent input, everyone is able to check if the transaction fragment is
+still valid, and the initiating party can cancel the swap proposal by simply spending input to themselves in a separate transaction, burning its key image and preventing any of the
+broadcasted transaction fragments from being used in a valid transaction.
```

Comment:
Maybe worth noting that without escrow, the burden is on the client to provide UX that informs the user how much of their funds are "pending" - this is similar to the way clients work today with transactions. The protocol layer does not report on "pending" or consider funds in a pending state.